lib/core/string.toit

```diff
@@ -1334,6 +1334,18 @@ abstract class string implements Comparable io.Data:
   write-to-byte-array_ byte-array/ByteArray start end dest-index:
     #primitive.core.string-write-to-byte-array
 
+  byte-size -> int:
+    return size
+
+  byte-slice from to/int -> io.Data:
+    if not 0 <= from <= to <= byte-size: throw "OUT_OF_BOUNDS"
+    if this is String_: return StringByteSlice_ (this as String_) from to
```

Comment:
Reverse the top two lines to avoid duplicating the bounds check in the common case.

Reply:
done.